### PR TITLE
[mono] Track .dotnet runtime version

### PR DESCRIPTION
**Pull request by EgorBo**
Local `.dotnet` folder we use for experiments, may contain more than one runtime (e.g. if global.json was updated recently) so we need to use the correct one.

### CODE REVIEW

**Comment by akoeplinger** [comment]
Given that SDKs don't update that often couldn't we also just error out when multiple are present in `.dotnet` and tell the user to run `git clean -xdff` (or delete the old one manually)?

**Comment by jkoritzinsky** [comment]
There's a much simpler way to get the version of the runtime bundled with the installed SDK. The `BundledNETCoreAppPackageVersion` MSBuild property will have the version of the runtime that was bundled with the installed SDK.

**Comment by EgorBo** [comment]
@jkoritzinsky any idea how to use it? it's empty for mono.proj 🙁

**Comment by jkoritzinsky** [comment]
You have to reference the Microsoft.NET.Sdk. So easiest way to do it would probably be make `mono,proj` use the [Microsoft.Build.NoTargets SDK](https://github.com/microsoft/MSBuildSdks/tree/master/src/NoTargets). That SDK includes the `Microsoft.NET.Sdk` SDK under the hood to support opening the project in VS, so the property should be available.

**Comment by EgorBo** [comment]
> You have to reference the Microsoft.NET.Sdk. So easiest way to do it would probably be make `mono,proj` use the [Microsoft.Build.NoTargets SDK](https://github.com/microsoft/MSBuildSdks/tree/master/src/NoTargets). That SDK includes the `Microsoft.NET.Sdk` SDK under the hood to support opening the project in VS, so the property should be available.

@jkoritzinsky thanks!
However I don't understand why on windows I get: "3.1.2" (which is system dotnet) 
I run it like this:

```
..\..\.dotnet\dotnet.exe msbuild /t:foo mono.proj
```
(foo is a test target to print `BundledNETCoreAppPackageVersion`)
However, it all works on macos/linux.
It feels like Windows just ignores my path to `.dotnet` and always uses system dotnet.

**Comment by jkoritzinsky** [comment]
That's probably because of our `global.json`'s "SDK version" is 3.1.100, but we pull down a newer SDK. As a result, if you have a 3.x sdk installed, we don't roll-forward to the prerelease. Try using the `dotnet.cmd` and `dotnet.sh` scripts in the root of the repository if possible or set `DOTNET_MULTILEVEL_LOOKUP=0` in your environment (one of the steps that the `dotnet.cmd/sh` scripts do).

File: src/mono/mono.proj [hunk 1]
```diff
@@ -259,6 +258,8 @@
   </Target>
 
   <Target Name="ValidateLocalDotnet">
+    <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalDotnetDir)\shared\Microsoft.NETCore.App').get_Length()) > 1" Text="$(LocalDotnetDir) contains more than one sdk" />
+    <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalMonoDotnetDir)\shared\Microsoft.NETCore.App').get_Length()) > 1" Text="$(LocalMonoDotnetDir) contains more than one sdk" />
```
Comment (akoeplinger) [inline src/mono/mono.proj]:
```suggestion
    <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalDotnetDir)\shared\Microsoft.NETCore.App').Length) > 1" Text="$(LocalDotnetDir) contains more than one SDK, please delete unused ones." />
    <Error Condition="$([System.IO.Directory]::GetDirectories('$(LocalMonoDotnetDir)\shared\Microsoft.NETCore.App').Length) > 1" Text="$(LocalMonoDotnetDir) contains more than one SDK, please delete unused ones." />
```

**Review by akoeplinger — APPROVED** [review]
LGTM with nits

**Comment by jkoritzinsky** [comment]
Why did you change this back to an error?

**Comment by EgorBo** [comment]
@jkoritzinsky couldn't manage to fix all the weird build issues I had after I set SDK in mono.proj (I am not super familiar with all the SDK/infrastructure tricks/variables yet so will revise it later) I guess that SDK overwrote some variables.

**Pull request merged**